File: .claude/skills/fixtures/core6-mechanical/typescript/change-signature/after.ts
interface RequestOptions {
  timeout?: number;
  retries?: number;
}

function fetchData(url: string, options?: RequestOptions): Promise<string> {
  return Promise.resolve(`data from ${url}`);
}

async function main(): Promise<void> {
  const result1 = await fetchData('https://api.example.com/users');
  const result2 = await fetchData('https://api.example.com/posts');
  const result3 = await fetchData('https://api.example.com/comments');
  console.log(result1, result2, result3);
}

main();

File: .claude/skills/fixtures/core6-mechanical/go/change-signature/after.go
package main

import (
	"context"
	"fmt"
)

// FetchData retrieves data from the given URL using the provided context.
// context.Context is always the first parameter by Go convention.
func FetchData(ctx context.Context, url string) (string, error) {
	return fmt.Sprintf("data from %s", url), nil
}

func main() {
	ctx := context.Background()
	d1, _ := FetchData(ctx, "https://api.example.com/users")
	d2, _ := FetchData(ctx, "https://api.example.com/posts")
	d3, _ := FetchData(ctx, "https://api.example.com/comments")
	fmt.Println(d1, d2, d3)
}
